### PR TITLE
Add test to check access, Gid, Uid of /opt/app-root/lib and /opt/app-root/share  (ODS-486)

### PR DESCRIPTION
**Pull request by pranavsuryaw**
Signed-off-by: Pranav Suryawanshi <prsuryaw@redhat.com>

### CODE REVIEW

**Comment by pranavsuryaw** [comment]
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/580/console
Result : Passed

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    ...    ${list_of_images}
+    ...    s2i-minimal-notebook
+    ...    s2i-generic-data-science-notebook
+    ...    pytorch
+    ...    tensorflow
+    ${permissions} =    Create List    0775    0    1001
+    ${path} =    Create List    lib    share
+    FOR    ${img}    IN    @{list_of_images}
+        Verify The Permissions Of Folder In Image    ${img}    ${path}    ${permissions}
+    END
```
Comment (bdattoma) [inline tests/Tests/500__jupyterhub/test-folder-permissions.robot:30]: could you re-format the test case to be more "high-level" as descripted in [style guide](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit#heading=h.4a2fb94fltk2) ?

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    [Tags]    ODS-486
+    ...       Tier2
+    @{list_of_images} =    Create List
+    Append To List
+    ...    ${list_of_images}
+    ...    s2i-minimal-notebook
+    ...    s2i-generic-data-science-notebook
+    ...    pytorch
+    ...    tensorflow
+    ${permissions} =    Create List    0775    0    1001
```
Comment (jgarciao) [inline tests/Tests/500__jupyterhub/test-folder-permissions.robot:26]: ${permissions} and ${path} can also be in the ***Variables*** section. 
 
 Consider renaming them to ${expected_permissions} and ${folders_to_check}

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+...                    pytorch    tensorflow
+@{EXPECTED_PERMISSIONS} =       0775    0    1001
+@{FOLDER_TO_CHECK} =            /opt/app-root/lib    /opt/app-root/share
+
+*** Test Cases ***
+Verify Folder Permissions
+    [Documentation]    It checks Access, Gid, Uid of /opt/app-root/lib and /opt/app-root/shares
+    [Tags]    ODS-486
+    ...       Tier2
+    FOR    ${img}    IN    @{LIST_OF_IMAGES}
```
Comment (jgarciao) [inline tests/Tests/500__jupyterhub/test-folder-permissions.robot:24]: `Verify Folder Permissions For Images ` @{LIST_OF_IMAGES}

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot (hunk 2 above)
Comment (jgarciao) [reply tests/Tests/500__jupyterhub/test-folder-permissions.robot]: Done

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot [hunk 4]
```diff
@@ -156,11 +156,11 @@ Spawn Notebook With Arguments  # robocop: disable
          ${is_launcher_selected} =  Run Keyword And Return Status  JupyterLab Launcher Tab Is Selected
          Run Keyword If  not ${is_launcher_selected}  Open JupyterLab Launcher
          Open With JupyterLab Menu  File  New  Notebook
-         Sleep  1
+         Sleep  3
          Maybe Close Popup
          Close Other JupyterLab Tabs
          Maybe Close Popup
-         Sleep  1
+         Sleep  3
```
Comment (jgarciao) [inline tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot:163]: Better `Wait Until ...`

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+Resource            ../../Resources/Common.robot
+Resource            ../../Resources/Page/ODH/JupyterHub/ODHJupyterhub.resource
+
+Library             JupyterLibrary
+
+Suite Setup         Load Spawner Page
+Suite Teardown      End Web Test
+
+*** Variables ***
+@{LIST_OF_IMAGES} =    s2i-minimal-notebook    s2i-generic-data-science-notebook
```
[the remark this answers is not in the document]
Comment (lugi0) [reply tests/Tests/500__jupyterhub/test-folder-permissions.robot:14]: `minimal-gpu`

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+...                    pytorch    tensorflow
+@{EXPECTED_PERMISSIONS} =       0775    0    1001
+@{FOLDER_TO_CHECK} =            /opt/app-root/lib    /opt/app-root/share
+
+*** Test Cases ***
+Verify Folder Permissions
+    [Documentation]    It checks Access, Gid, Uid of /opt/app-root/lib and /opt/app-root/shares
```
[the remark this answers is not in the document]
Comment (pranavsuryaw) [reply tests/Tests/500__jupyterhub/test-folder-permissions.robot]: Done

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot (hunk 5 above)
[the remark this answers is not in the document]
Comment (pranavsuryaw) [reply tests/Tests/500__jupyterhub/test-folder-permissions.robot:14]: Done

File: tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot (hunk 4 above)
Comment (pranavsuryaw) [reply tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot]: Done

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot (hunk 3 above)
Comment (pranavsuryaw) [reply tests/Tests/500__jupyterhub/test-folder-permissions.robot]: Done

File: tests/Tests/500__jupyterhub/test-folder-permissions.robot (hunk 1 above)
Comment (pranavsuryaw) [reply tests/Tests/500__jupyterhub/test-folder-permissions.robot]: Read it and made changes accordingly

**Review by lugi0 — APPROVED** [review]
lgtm

**Review by jgarciao — APPROVED** [review]
lgtm

**Review by bdattoma — APPROVED** [review]
lgtm

**Pull request merged**